packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -208,17 +238,43 @@ function OrgSelect({
       return search.test(datum.orgName);
     });
 
-    return filtered;
+    setFilteredData(filtered);
   };
 
-  const renderedData = onChangeProp ? consumerFilteredData : filterData();
+  const didMount = useRef(false);
+  useEffect(() => {
+    // allow the user to provide filtered data
+    if (onChangeProp) {
+      return;
+    }
+
+    // manage serverside filtering for a global user
+    if (admin) {
+      // don't fetch data initially, only when filtering
+      if (!didMount.current) {
+        didMount.current = true;
+        return;
+      }
+
+      filterDataAsAdmin();
```

Comment:
We need to debounce this as we currently do in ContextSwitcher. Otherwise every character input is going to create an API request.

Looks like we debounce for 300ms in mms.

Reply:
Also the behavior we have in ContextSwitcher when `value === ''` we reset back to the original data for the user, so the passed in `data` value.